unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/PlacesSubSection/FavoritesSubSection/FavoriteSubSectionComponentController.cs

```diff
@@ -0,0 +1,189 @@
+using Cysharp.Threading.Tasks;
+using DCL;
+using DCL.Helpers;
+using DCL.Tasks;
+using DCLServices.PlacesAPIService;
+using DCLServices.WorldsAPIService;
+using System.Collections.Generic;
+using ExploreV2Analytics;
+using MainScripts.DCL.Controllers.HotScenes;
+using System;
+using System.Threading;
+using Environment = DCL.Environment;
+
+public class FavoriteSubSectionComponentController : IFavoriteSubSectionComponentController
+{
+    private readonly IFavoriteSubSectionComponentView view;
+    private readonly IPlacesAPIService placesAPIService;
+    private readonly IWorldsAPIService worldsAPIService;
+    private readonly IUserProfileBridge userProfileBridge;
+    private readonly IExploreV2Analytics exploreV2Analytics;
+    private readonly IPlacesAnalytics placesAnalytics;
+    private readonly DataStore dataStore;
+
+    public event Action OnCloseExploreV2;
+
+    private CancellationTokenSource minimalListCts;
+    private CancellationTokenSource fullFavoriteListCts;
+
+    public FavoriteSubSectionComponentController(
+        IFavoriteSubSectionComponentView view,
+        IPlacesAPIService placesAPIService,
+        IWorldsAPIService worldsAPIService,
+        IUserProfileBridge userProfileBridge,
+        IExploreV2Analytics exploreV2Analytics,
+        IPlacesAnalytics placesAnalytics,
+        DataStore dataStore)
+    {
+        this.view = view;
+        this.placesAPIService = placesAPIService;
+        this.worldsAPIService = worldsAPIService;
+        this.userProfileBridge = userProfileBridge;
+        this.exploreV2Analytics = exploreV2Analytics;
+        this.placesAnalytics = placesAnalytics;
+        this.dataStore = dataStore;
+
+        view.OnRequestAllPlaces += RequestAllFavoritePlaces;
+        view.OnRequestAllWorlds += RequestAllFavoriteWorlds;
+        view.OnPlaceInfoClicked += OpenPlaceDetailsModal;
+        view.OnWorldInfoClicked += OpenWorldDetailsModal;
+        view.OnVoteChanged += ChangeVote;
+        view.OnPlaceJumpInClicked += JumpInToPlace;
+        view.OnWorldJumpInClicked += JumpInToWorld;
+        view.OnPlaceFavoriteChanged += ChangePlaceFavorite;
+        view.OnRequestFavorites += RequestFavorites;
+    }
+
+    private void ChangeVote(string placeId, bool? isUpvote)
+    {
+        if (userProfileBridge.GetOwn().isGuest)
+            dataStore.HUDs.connectWalletModalVisible.Set(true);
+        else
+        {
+            if (isUpvote != null)
+            {
+                if (isUpvote.Value)
+                    placesAnalytics.Like(placeId, IPlacesAnalytics.ActionSource.FromFavorites);
+                else
+                    placesAnalytics.Dislike(placeId, IPlacesAnalytics.ActionSource.FromFavorites);
+            }
+            else
+                placesAnalytics.RemoveVote(placeId, IPlacesAnalytics.ActionSource.FromFavorites);
+
+            placesAPIService.SetPlaceVote(isUpvote, placeId, default).Forget();
```

Comment:
don't really need to cancel this, it's async just because it might take time but there are no scenarios in which we need to cancel, also we can have multiple parallel requests